providers/aws/services/vpc/vpc_service.py

```diff
@@ -0,0 +1,308 @@
+import json
+import threading
+from dataclasses import dataclass
+
+from lib.logger import logger
+from providers.aws.aws_provider import generate_regional_clients
+
+
+################## VPC
+class VPC:
+    def __init__(self, audit_info):
+        self.service = "ec2"
+        self.session = audit_info.audit_session
+        self.audited_account = audit_info.audited_account
+        self.regional_clients = generate_regional_clients(self.service, audit_info)
+        self.vpcs = []
+        self.vpc_peering_connections = []
+        self.vpc_endpoints = []
+        self.vpc_endpoint_services = []
+        self.__threading_call__(self.__describe_vpcs__)
+        self.__threading_call__(self.__describe_vpc_peering_connections__)
+        self.__threading_call__(self.__describe_vpc_endpoints__)
+        self.__threading_call__(self.__describe_vpc_endpoint_services__)
+        self.__describe_flow_logs__()
+        self.__describe_route_tables__()
+        self.__describe_vpc_endpoint_service_permissions__()
+
+    def __get_session__(self):
+        return self.session
+
+    def __threading_call__(self, call):
+        threads = []
+        for regional_client in self.regional_clients.values():
+            threads.append(threading.Thread(target=call, args=(regional_client,)))
+        for t in threads:
+            t.start()
+        for t in threads:
+            t.join()
+
+    def __describe_vpcs__(self, regional_client):
+        logger.info("VPC - Describing VPCs...")
+        try:
+            describe_vpcs_paginator = regional_client.get_paginator("describe_vpcs")
+            for page in describe_vpcs_paginator.paginate():
+                for vpc in page["Vpcs"]:
+                    self.vpcs.append(
+                        VPCs(
+                            vpc["VpcId"],
+                            vpc["IsDefault"],
+                            vpc["CidrBlock"],
+                            regional_client.region,
+                        )
+                    )
+        except Exception as error:
+            logger.error(
+                f"{regional_client.region} -- {error.__class__.__name__}: {error}"
```

Comment:
The exception line number is missing here

Reply:
We have to include it in a global function

Reply:
Done